controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/Upgrader.java

```diff
@@ -92,8 +92,10 @@ public void maintain() {
         return reversed;
     }
 
-    /** Returns a list of all applications */
-    private ApplicationList applications() { return ApplicationList.from(controller().applications().asList()); }
+    /** Returns a list of all applications, except those which are pinned — these should not be manipulated by the Upgrader */
+    private ApplicationList applications() {
+        return ApplicationList.from(controller().applications().asList()).notPinned();
```

Comment:
Consider moving all filtering to the `upgrade` method, together with the others. Easier to read when all filtering happens in one place.

Reply:
It's not relevant just for upgrades, but also cancellation. That's why I put it here.